### PR TITLE
feat: harmonization grid in variable overview

### PR DESCRIPTION
**Pull request by connoratrug**
For each variable fetch the status for each cohort ( and repeat)
show in grid

paged, page size 10 , load more on user request ( performance work around)

### CODE REVIEW

File: apps/catalogue/src/store/actions.js [hunk 1]
```diff
@@ -221,59 +222,33 @@ export default {
...
-    const filter = variables.length
-      ? {
-          filter: { toVariable: { equals: variables } },
-        }
-      : {};
+      },
+    };
 
-    const resp = await request("graphql", query, filter).catch((e) =>
+    const resp = await request("graphql", mappings, { filter }).catch((e) =>
```
Comment (sidohaakma) [inline apps/catalogue/src/store/actions.js:249]: That cleans up nicely!

File: apps/catalogue/src/views/HarmonizationRow.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  methods: {
+    ...mapActions(["fetchMappings"]),
+    getCellClass(cohort) {
+      return this.cohortMappings ? this.getMatchStatus(cohort) : null;
+    },
+    async fetchData() {
+      this.cohortMappings = await this.fetchMappings(this.variable);
+    },
+    getMatchStatus(cohort) {
+      if (this.variable.repeats) {
```
Comment (sidohaakma) [inline apps/catalogue/src/views/HarmonizationRow.vue:37]: Maybe extract this in a service or something or to the getters maybe? Later on for testing purposes it is easier when it is not in the vue component I guess? What do you think?

**Comment by sidohaakma** [comment]
[Add harmonisation grid](https://trello.com/c/lVlcy5RW/99-add-harmonisation-grid)

File: apps/catalogue/src/views/HarmonizationRow.vue (hunk 2 above)
Comment (mswertz) [reply apps/catalogue/src/views/HarmonizationRow.vue:37]: I would extract it when I then indeed would need it seperate for testing. Interesting to observe how we seem to differ in our preferences. My preference is to keep things that belong together in one file as much as I can. You seem more at ease to working in multiple files. I think there is no objective truth so I propose this is up to the developer writing the code :-)

Comment (connoratrug) [reply apps/catalogue/src/views/HarmonizationRow.vue:37]: i agree with both, 'keep things that belong together in one file' the same type of calc ( not the same) is used in other component. should probably be moved into service then these two functions are kept together. This PR is more explorative, trying to explore if the work around for missing argg data by paging the table work from a use perspective

**Review by sidohaakma — APPROVED** [review]

**Pull request merged**